src/utils/index.js

```diff
@@ -155,6 +155,48 @@ const isPristine = (initialValues, currentFormValues, runKey) => {
   });
 };
 
+const getOptionsData = (options) => {
+  if (!options) {
+    return [];
+  }
+  const { data } = options;
+  if (!data || !data.actions) {
+    return [];
+  }
+  return data.actions.POST;
+};
+
+const parseCourseTypeOptions = (typeOptions) => {
+  const courseTypeOptionsArray = [{ label: 'Select enrollment track', value: '' }];
+  const courseRunTypeOptions = {};
+  const entitlementUUIDS = [];
+  for (let i = typeOptions.length - 1; i >= 0; i -= 1) {
+    const type = typeOptions[i];
+    courseTypeOptionsArray.push({
+      label: type.name, value: type.uuid,
+    });
+    const courseRunTypeArray = [{ label: 'Select enrollment track', value: '' }];
+    for (let j = type.course_run_types.length - 1; j >= 0; j -= 1) {
+      const courseRunType = type.course_run_types[j];
+      courseRunTypeArray.push({
+        label: courseRunType.name, value: courseRunType.uuid,
+      });
+    }
+    courseRunTypeOptions[type.uuid] = courseRunTypeArray;
+    // This should be changed to remove the part about AUDIT_TRACK once we stop
+    // creating draft Audit entitlements. Right now, this is used to decide if we
+    // show the price on the course create/edit form.
+    if (type.entitlement_types.length && type.entitlement_types[0] !== AUDIT_TRACK.key) {
```

Comment:
`[0]` seems brittle. Maybe also add `length == 1` instead of just `>0`? But I get that you are hoping for a short lifetime for this code.